controllers/account/account_controller.go

```diff
@@ -106,12 +114,47 @@ func (r *AccountReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		return reconcile.Result{}, err
 	}
 
+	// Retrieve a list of accounts with region enablement in progress for supported Opt-In regions
+	accountList := &awsv1alpha1.AccountList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(awsv1alpha1.AccountCrNamespace),
+	}
+
+	if err = r.Client.List(context.TODO(), accountList, listOpts...); err != nil {
+		if k8serr.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// since it's not possible to filter on custom field values when listing using the golang client
+	// manual filtering of accounts opting-in is required to ensure the account limit is not reached
+	numberOfAccountsOptingIn := 0
+	for _, account := range accountList.Items {
+		if account.Status.State == "OptingInRegions" {
+			numberOfAccountsOptingIn += 1
+		}
+	}
+
 	configMap, err := utils.GetOperatorConfigMap(r.Client)
 	if err != nil {
 		log.Error(err, "Failed retrieving configmap")
 		return reconcile.Result{}, err
 	}
 
+	var isEnabled = false
```

Comment:
Can we rename this var to something more descriptive?

Maybe this whole function could be abstracted?